### ISSUE TITLE
Metrics are flawed on negative samples

### ISSUE THREAD

**Issue by LM239** [Sjyhne/MapAI-Competition#4]
Both the metrics for  IoU and BIoU are incapable of evaluating samples with no buildings because True Negatives are not accounted for. This also leads to very unstable metrics for samples with almost exclusively background.

For a sample with no buildings, any prediction, regardless of its accuracy, will give a score of 0. This happens because the metrics are only calculated for the positive class, but not the negative class.

Since True Negatives and False Positives are important measures of performance, I highly encourage you to change how you evaluate predictions. I hope you see the urgency here; a competition with flawed metrics is not fair and can't be taken seriously.

Please implement class-wise metrics following Keras' [MeanIOU](https://keras.io/api/metrics/segmentation_metrics/#meaniou-class) metric. Its documentation states:

> Note that this class first computes IoUs for all individual classes, then returns the mean of these values.

Additionally (in a class-wise implementation), both IoU and BIoU should ignore a class (or default to 1) if its target has no positive examples.

**Comment by Sjyhne** [comment]
Thank you for raising this issue. I've now updated the main repository with functionality for class-wise IoU and BIoU evaluation. Hopefully, this solves the problem.

I've also added a description for updating to the newest version of the fork - and functionality for checking your current fork version against the newest on Github. 

Please close the issue if you're satisfied with the solution.

**Comment by LM239** [comment]
Thank you! But BIoU is still having issues, because mboundary_iou can be empty, thereby giving NaN as BIoU. Consider using:

```
def class_wise(arr: np.array, c: int) -> np.array:
    return arr == c

def biou(dt, gt, dilation_ratio=0.02):

    mboundary_iou = []

    for c in range(2):

        target = class_wise(gt, c)
        if not np.any(target):
            continue

        prediction = class_wise(dt, c)

        gt_boundary = _mask_to_boundary(target.astype(np.uint8), dilation_ratio)
        dt_boundary = _mask_to_boundary(prediction.astype(np.uint8), dilation_ratio)
        intersection = ((gt_boundary * dt_boundary) > 0).sum()
        union = ((gt_boundary + dt_boundary) > 0).sum()
        if union == 0 or intersection == 0:
            boundary_iou = 0
        else:
            boundary_iou = (intersection / union)

        mboundary_iou.append(boundary_iou)
        
    return np.asarray(mboundary_iou).mean()
```

This corrects the order of the input parameters in the function definition, uses uint8 for _mask_to_boundary and does not return NaN.

**Comment by Sjyhne** [comment]
Ah, yes. Thank you for correcting my implementation! The update is now pushed.
